crates/router/src/connector/netcetera/netcetera_types.rs

```diff
@@ -716,11 +721,24 @@ impl
             bill_addr_state: billing_address
                 .address
                 .as_ref()
-                .and_then(|add| add.state.clone()),
+                .and_then(|add| add.to_state_code_as_optional().transpose())
```

Comment:
We need not call `to_state_code_as_optional` here if it's already sent as state code, we should check and call this fn only if required (as `state` in our request is a String and any value can be sent there)